### PR TITLE
Rework cmd_run to use new driver and config systems

### PR DESCRIPTION
**Pull request by jakelang**

### CODE REVIEW

File: chisel/src/main.rs [hunk 1]
```diff
@@ -6,65 +6,102 @@ mod driver;
...
+        .arg(
+            Arg::with_name("MODULES")
+                .short("m")
+                .long("modules")
+                .takes_value(true)
+                .multiple(true)
+                .help("Selects modules to use"),
+        )
+        .arg(Arg::with_name("FILE").help("File to chisel"))
+        .arg(
```
Comment (axic) [inline chisel/src/main.rs]: Don't wanna nitpick too much but would prefer to not include oneliner specific options in this PR, like `OUTPUT_PATH`, `MODULE_OPIONS` and `OUTPUT_MODE`.

File: chisel/src/main.rs [hunk 2]
```diff
@@ -6,65 +6,102 @@ mod driver;
...
+        )
+        .arg(
+            Arg::with_name("MODULES")
+                .short("m")
+                .long("modules")
+                .takes_value(true)
+                .multiple(true)
+                .help("Selects modules to use"),
+        )
+        .arg(Arg::with_name("FILE").help("File to chisel"))
```
Comment (axic) [inline chisel/src/main.rs]: This doesn't have a short or long option, does this mean it is a leaf argument?

Comment (axic) [reply chisel/src/main.rs]: If so perhaps it should be last before the help message.

File: chisel/src/main.rs [hunk 3]
```diff
@@ -6,65 +6,102 @@ mod driver;
...
                         .short("c")
                         .long("config")
-                        .help("Sets a custom configuration file")
-                        .value_name("CONF_FILE")
+                        .help("Sets the configuration file in config-driven mode.")
+                        .value_name("PATH")
                         .takes_value(true),
                 ),
         )
+        .after_help("chisel runs in two primary modes: unix-style and config-driven.\n\nunix-style is invoked without a subcommand. \
```
Comment (axic) [inline chisel/src/main.rs]: Perhaps move this into the oneliner PR too.

File: chisel/src/main.rs [hunk 4]
```diff
@@ -6,65 +6,102 @@ mod driver;
...
-            process::exit(chisel_run(subcmd_matches))
+        ("run", args) => {
+            if let Some(opts) = args {
+                flags.apply(opts);
+            }
+
+            chisel_run(flags)
         }
-        _ => err_exit(ERR_NO_SUBCOMMAND),
+        // Run chisel in unix mode if no subcommand provided
```
Comment (axic) [inline chisel/src/main.rs]: Same here.

File: chisel/src/main.rs [hunk 5]
```diff
@@ -6,65 +6,102 @@ mod driver;
...
-};
+use cmd_run::chisel_run;
+use options::ChiselFlags;
 
 fn fail(code: i32, message: &str) -> ! {
     eprintln!("{}: {}", crate_name!(), message);
     process::exit(code);
 }
 
-#[cfg(feature = "binaryen")]
```
Comment (axic) [inline chisel/src/main.rs]: Hm was this a leftover?

File: chisel/src/main.rs (hunk 2 above)
Comment (jakelang) [reply chisel/src/main.rs]: It's an argument which is invoked without a flag. Will move to end.

File: chisel/src/main.rs (hunk 1 above)
Comment (jakelang) [reply chisel/src/main.rs]: Ok

File: chisel/src/main.rs [hunk 6]
```diff
@@ -6,50 +6,79 @@ mod driver;
...
+        )
+        .arg(
+            Arg::with_name("DEBUG_MESSAGES")
+                .short("d")
+                .long("debug")
+                .help("Enables debug messages")
                 .global(true),
         )
+        .arg(
+            Arg::with_name("MODULES")
```
Comment (axic) [inline chisel/src/main.rs]: Is this part of oneliner too ?

File: chisel/src/main.rs (hunk 3 above)
Comment (jakelang) [reply chisel/src/main.rs]: It's a description for both, I think this should be no big issue.

File: chisel/src/cmd_run.rs [hunk 7]
```diff
@@ -1,315 +1,150 @@
...
+                Some("hex") => ruleset.write("hex"),
+                _ => panic!("CLI parser ensures value can only be one of the above"),
+            };
+            (ruleset_name, write_result)
+        })
+        .filter_map(|(name, result)| {
+            if result.is_err() {
+                Some((name, result.unwrap_err()))
+            } else {
+                //
```
Comment (axic) [inline chisel/src/cmd_run.rs]: Oops, stray comment ?

**Pull request merged**